### PR TITLE
Fix connection removal bug

### LINKED ISSUE #9126: MetaMesk does not always send the 'accountsChanged' event on v8.0.5
**Describe the bug**
1. MetaMesk does not always send the 'accountsChanged' event on v8.0.5
2. When changing the account address on MetaMask, the DApp cannot get the new address. If the DApp submit a tx, the MetaMask get a console log:
```MetaMask: 'eth_accounts' unexpectedly updated accounts. Please report this bug.```
3. the bug happens only when upgrading to v8.0.5

**To Reproduce (REQUIRED)**

1. Go to mcdex.io
2. Connect multi-accounts
3. Open to AMM page
4. switch between accounts
5. if the address on mcdex.io does not change corresponding, do a deposit transaction, get console log:
```MetaMask: 'eth_accounts' unexpectedly updated accounts. Please report this bug.```

**Expected behavior**
The account on mcdex.io changes when the user selects different accounts on MetaMask.

**Screenshots**

**Browser details (please complete the following information):**
 - OS: Windows 10
 - Browser [e.g. Chrome Version  84.0.4147.105
 - MetaMask Version [e.g. 8.0.5]

**Additional context (Error Messages, etc.)**

### PR DESCRIPTION
**Pull request by rekmarks**
Fixes what is effectively a type error that caused _all_ connections for a given origin to be removed whenever one connection was removed.

Fixes #9126

### CODE REVIEW

**Review by brad-decker — DISMISSED** [review]

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -1722,7 +1722,7 @@ export default class MetamaskController extends EventEmitter {
 
     delete connections[id]
 
-    if (Object.keys(connections.length === 0)) {
+    if (Object.keys(connections.length) === 0) {
```
Comment (Gudahtt) [inline app/scripts/metamask-controller.js:1725]: I think the paren is still in the wrong spot :sweat_smile: 

```suggestion
    if (Object.keys(connections).length === 0) {
```

**Review by Gudahtt — CHANGES_REQUESTED** [review]

File: app/scripts/metamask-controller.js (hunk 1 above)
Comment (rekmarks) [reply app/scripts/metamask-controller.js:1725]: 😱

Comment (rekmarks) [reply app/scripts/metamask-controller.js]: 0779825

😑

Comment (Gudahtt) [reply app/scripts/metamask-controller.js]: The third time's the charm hopefully?

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**